File: C_Sharp/Framework.Tests/LazyLoadTests.cs
using System.Collections.Generic;
using System.Linq;
using Framework.Data_Manipulation;
using Framework.LazyLoad;
using Framework.Tests.LazyLoad;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Framework.Tests
{
    [TestClass]
    public class LazyLoadTests
    {

        /****************************/
        //
        //      PLEASE READ 'LazyLoad_README.txt'
        //
        /****************************/
        [TestMethod]
        public void TestLazyLoadProducts()
        {
            LazyLoader<ProductDomain, ProductDomain.Criteria> loader = new ProductLazyLoader();
            IList<ProductDomain> list = new LazyLoadList<ProductDomain, ProductDomain.Criteria>(loader);
            IBaseMapper mapper = null;

            list.Add(new ProductDomain(mapper, ProductDomain.Criteria.SearchById(2)));
            list.Add(new ProductDomain(mapper, ProductDomain.Criteria.SearchById(4)));
            list.Add(new ProductDomain(mapper, ProductDomain.Criteria.SearchById(5)));

            ProductDomain product = list.FirstOrDefault(x => x.Id == 2);

            Assert.IsTrue(string.IsNullOrEmpty(product.Description));       //Description is not loaded
            Assert.AreEqual("Product two", list[0].Description);
            Assert.AreEqual("Product four", list[1].Description);
            Assert.AreEqual("Product five", list[2].Description);
            Assert.IsFalse(string.IsNullOrEmpty(product.Description));      //Description is now loaded
        }
    }
}

File: Android/app/src/test/java/com/LazyLoadTests.java
package com;

import com.DataManipulation.BaseMapperInterfaces.IBaseMapper;
import com.LazyLoad.LazyLoadList;
import com.LazyLoad.LazyLoader;

import java.util.List;

import com.LazyLoad.ProductDomain;
import com.LazyLoad.ProductLazyLoader;

import static org.junit.Assert.assertEquals;

/**
 * Created by aiko on 4/8/17.
 */

public class LazyLoadTests {
    /****************************/
    //
    //      PLEASE READ 'LazyLoad_README.txt'
    //
    /****************************/
    @org.junit.Test
    public void TestLazyLoadProducts()
    {
        LazyLoader<ProductDomain, ProductDomain.Criteria> loader = new ProductLazyLoader();
        List<ProductDomain> list = new LazyLoadList<ProductDomain, ProductDomain.Criteria>(loader);
        IBaseMapper mapper = null;

        list.add(new ProductDomain(mapper, new ProductDomain.Criteria(2)));
        list.add(new ProductDomain(mapper, new ProductDomain.Criteria(4)));
        list.add(new ProductDomain(mapper, new ProductDomain.Criteria(5)));

        assertEquals("Product two", list.get(0).Description);
        assertEquals("Product four", list.get(1).Description);
        assertEquals("Product five", list.get(2).Description);
    }
}
